main.go

```diff
@@ -208,6 +217,8 @@ func main() {
 		os.Exit(1)
 	}
 
+	go addDiscoverySecretWatch(context.TODO(), mgr, uncachedClient)
```

Comment:
context.TODO() should be context.Background() instead. They function identically, but TODO() is designed to be a context that's for updating later when still in development